plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteClient.java

```diff
@@ -391,26 +392,36 @@ public JdbcOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
         }
     }
 
-    private String buildCreateSql(SchemaTableName schemaTableName, List<String> columns, Map<String, Object> tableProperties)
+    private String buildCreateSql(SchemaTableName schemaTableName, List<String> columns, List<String> columnNames, Map<String, Object> tableProperties)
     {
         ImmutableList.Builder<String> columnDefinitions = ImmutableList.builder();
         columnDefinitions.addAll(columns);
 
         List<String> primaryKeys = IgniteTableProperties.getPrimaryKey(tableProperties);
         checkArgument(primaryKeys.size() < columns.size(), "Ignite table must have at least one non PRIMARY KEY column.");
-        if (primaryKeys.isEmpty()) {
-            columnDefinitions.add(quoted(IGNITE_DUMMY_ID) + " VARCHAR NOT NULL");
-            primaryKeys = ImmutableList.of(IGNITE_DUMMY_ID);
-        }
-        columnDefinitions.add("PRIMARY KEY (" + join(", ", primaryKeys.stream().map(this::quoted).collect(joining(", "))) + ")");
 
         for (Map.Entry<String, Object> propertyEntry : tableProperties.entrySet()) {
```

Comment:
have changed the relative order between the `if (primaryKeys.isEmpty())` block and `for (Map.Entry<String, Object> propertyEntry : tableProperties.entrySet()) {` 
the latter only performs validation checks and this reorder doesnt affect correctness..

reordering is needed because without it, the `if (primaryKeys.isEmpty())` adds a dummy id column and will require special case while checking for primary key present in columnsName list.